backend/app/app.go

```diff
@@ -0,0 +1,93 @@
+package app
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+)
+
+// Instance defines the interface for app plugin instances.
+type Instance interface {
+	backend.CheckHealthHandler
+	backend.CallResourceHandler
+}
+
+// InstanceProviderFunc is the factory method for creating a new app plugin instance.
+type InstanceProviderFunc func(settings backend.AppInstanceSettings) (Instance, error)
+
+type Plugin interface {
+	ServeOpts() backend.ServeOpts
+	Serve() error
+}
+
+type plugin struct {
+	serveOpts backend.ServeOpts
+	im        instancemgmt.InstanceManager
+}
+
+func New(fn InstanceProviderFunc) Plugin {
+	if fn == nil {
+		panic("fn cannot be nil")
+	}
+
+	ip := NewInstanceProvider(func(settings backend.AppInstanceSettings) (instancemgmt.Instance, error) {
```

Comment:
I dont see how we could support multiple types with this PR. Maybe I'm missing something? But I think this restricts the `New` function to only work with Apps.